### PR TITLE
Open only one connection to a physical external device (#3888)

### PR DESCRIPTION
**Pull request by freddidierRTE**
- In release note :
  -  In chapter :  Features 
  -  Text : #3888 Open only one connection to a physical external device

### CODE REVIEW

File: src/docs/asciidoc/reference_doc/external_devices_service.adoc [hunk 1]
```diff
@@ -179,18 +179,19 @@ severities.
...
 
 == Configuration
 
 == Connexion Management
 
-OperatorFabric does automatically attempt to connect to enabled configured external devices at startup. If an enabled device
-is disconnected, OperatorFabric will try to reconnect it every 10 seconds (default value).
+OperatorFabric does automatically attempt to connect to enabled configured external devices at startup. If several external devices share the same host and port, only one connection will be open for the external devices and only one watchdog will be send for the external devices.
+
+If a connection is disconnected, OperatorFabric will try to reconnect it every 10 seconds (default value).
```
Comment (vlo-rte) [inline src/docs/asciidoc/reference_doc/external_devices_service.adoc:194]: I propose : "If a connection is disconnected" -> "If an external device is disconnected" ?

**Review by vlo-rte — CHANGES_REQUESTED** [review]
Please solve the code smells. And remove unused imports in DevicesController.java.

File: src/docs/asciidoc/reference_doc/external_devices_service.adoc [hunk 2]
```diff
@@ -179,18 +179,19 @@ severities.
...
-them differently (some people might be allowed to connect/disconnect devices but not change their configuration).
+them differently.
 
 == Configuration
 
 == Connexion Management
 
-OperatorFabric does automatically attempt to connect to enabled configured external devices at startup. If an enabled device
-is disconnected, OperatorFabric will try to reconnect it every 10 seconds (default value).
+OperatorFabric does automatically attempt to connect to enabled configured external devices at startup. If several external devices share the same host and port, only one connection will be open for the external devices and only one watchdog will be send for the external devices.
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply src/docs/asciidoc/reference_doc/external_devices_service.adoc:192]: ok done

File: src/docs/asciidoc/reference_doc/external_devices_service.adoc (hunk 1 above)
Comment (freddidierRTE) [reply src/docs/asciidoc/reference_doc/external_devices_service.adoc:194]: would be in this case external driver

File: services/external-devices/src/main/java/org/opfab/externaldevices/configuration/DataInitComponent.java [hunk 3]
```diff
@@ -114,11 +118,20 @@ private void initUserConfigurations(List<UserConfigurationData> userConfiguratio
...
             log.warn("{} {} device configuration: duplicate",FAILED_INIT_MSG, deviceConfigurationData.getId());
         }
     }
 
+    private void enableDevice(String deviceId) {
+        try {
+            deviceService.enableDevice(deviceId);
+        }
+        catch (Exception exc) {
+            log.warn("Impossible to enable device " + deviceId ,exc);
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply services/external-devices/src/main/java/org/opfab/externaldevices/configuration/DataInitComponent.java:132]: ok done

File: src/docs/asciidoc/reference_doc/external_devices_service.adoc (hunk 1 above)
Comment (freddidierRTE) [reply src/docs/asciidoc/reference_doc/external_devices_service.adoc:194]: done

**Review by vlo-rte — APPROVED** [review]

**Pull request merged**